### ISSUE TITLE
"Adding events to" appears multiple times in Google Calendar app settings

### ISSUE THREAD

**Issue by aspiers** [calcom/cal.com#13787]
### Issue Summary

I have 4 Google Calendar accounts connected to my cal.com account.  When I go to https://app.cal.com/apps/installed/calendar, I see two of these marked with the "Adding events to" label:

![image](https://github.com/calcom/cal.com/assets/100738/7795e1dc-1e82-4ad7-be58-43a8b2664b98)

I believe this label is meant to indicate the calendar which is selected as the default to add to events to, for event types which don't specify a different account as an override:

![image](https://github.com/calcom/cal.com/assets/100738/4fe4b0d2-a9c4-425f-a0c8-758e4c813f40)

In that case, there should only be one default, not two, so the fact that two "Adding events to" labels are simultaneously visible looks very much like a bug to me.

Assuming I'm right, I have a strong hunch that this bug is due to these calendars both having the same name, even though they come from different Google accounts.  This appears to be confirmed by both of the following observations:

1. Changing the default to a different calendar which has a unique name across all connected accounts results in only that uniquely named calendar having the "Adding events to" label.
2. Changing the default back to the non-unique name causes two "Adding events to" labels to appear on both instances of that name, as shown in the first screenshot above.

### Steps to Reproduce

1. Obtain two Google accounts which both have a calendar with the same name. 
2. Connect those accounts to cal.com
3. Check that the same calendar name can be seen in both accounts, similar to as shown in the first screenshot above.
4. Select either of those calendars as the default calendar to add events to.

### Actual Results

Two "Adding events to" labels are shown.

### Expected Results

Only the calendar selected as default should have an "Adding events to" label next to it.

### Technical details

- Google Chrome 119.0.6045.159 (Official Build) (64-bit)

### Evidence

See screenshots above.

**Comment by Pritam-sd-dev** [comment]
@aspiers 
We apologize for the inconvenience. Rest assured, the issue will be fixed as soon as possible.

**Comment by Pritam-sd-dev** [comment]
@aspiers 
i have tried to reproduce this bug but didnot find.
![Untitled design](https://github.com/calcom/cal.com/assets/113875582/82aaf295-806d-484b-8ef6-70b5abc1ba42)

**Comment by aspiers** [comment]
Could it be due to me sharing that calendar between accounts?

**Comment by Pritam-sd-dev** [comment]
@aspiers i didnot get "sharing that calender bw accounts"
can you please tell little more in breif or like you can provide video as reference for sharing calender bw accounts so that i can reproduce the bug.
thanks

**Comment by aspiers** [comment]
I have shared some of my calendars between Google accounts:

![image](https://github.com/calcom/cal.com/assets/100738/5765ec19-d9b5-4d9c-8682-d95a838456ee)

**Comment by aspiers** [comment]
Also @Pritam-sd-dev did your two calendars have exactly the same name?

**Comment by Pritam-sd-dev** [comment]
![Untitled design](https://github.com/calcom/cal.com/assets/113875582/16377d1f-544a-4adf-9064-7c3d6a905d2f)
found the bug.

**Comment by Pritam-sd-dev** [comment]
@PeerRich 
is this a bug or its default behaviour ?
if it is a bug. what should be the final behaviour.

**Comment by aspiers** [comment]
Now that I understand what's happening it makes sense, but it is confusing at first so ideally we could think of a way to make it clearer.

**Comment by emrysal** [comment]
@PeerRich (@aspiers) It's a good question actually if this is a bug - I've managed to reproduce this by creating a shared calendar with multiple accounts and then connect both accounts to the same shared calendar - the following code powers this indicator, which is actually the same calendar:

```jsx
destination={cal.externalId === props.destinationCalendarId}
```

I would argue this can be considered correct and I don't think it requires a fix.

**Comment by aspiers** [comment]
Yeah I think we're all saying the same thing: it's correct, but definitely confusingly non-intuitive.  Is there perhaps a simple text tweak which can be made to clarify?  E.g. if more than one "Adding events to" is displayed on the page, also display some clarifying text (perhaps in a tooltip) which explains that they are the same calendar but accessible from different accounts?  A UX expert should wade in here.
